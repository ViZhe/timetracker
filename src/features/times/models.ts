
export interface ITimesData {
  description: string;
  duration: string;
  key: string;
  tags: string[];
  timeStart: string;
  timeEnd: string | null;
}

export interface ITimesState {
  data: ITimesData[];
  isLoading: boolean;
}
