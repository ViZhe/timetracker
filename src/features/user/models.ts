
export interface IUserState {
  name: string | null;
}
