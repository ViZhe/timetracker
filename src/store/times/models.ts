
export interface ITimesData {
  description: string;
  duration: string;
  key: string;
  tags: string[];
  timeStart: Date | string;
  timeEnd: Date | string;
}

export interface ITimesState {
  data: ITimesData[];
  isLoading: boolean;
}
