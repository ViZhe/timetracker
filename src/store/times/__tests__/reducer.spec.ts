
import {
  reducer,
  timesActions as actions,
} from '../';
import { ITimesData, ITimesState } from '../models';


describe('Times Reducer', () => {
  const getInitialState = (initialState?: Partial<ITimesState>) => (
    reducer(initialState as ITimesState, {} as any)
  );

  describe('initial state', () => {
    const initialState = getInitialState();

    it('should match a snapshot', () => {
      expect(initialState).toMatchSnapshot();
    });
  });

  describe('adding times entry', () => {
    const initialState = getInitialState();

    it('should add a new times entry as the first element', () => {
      const payload:ITimesData = {
        description: 'Test dedescription',
        duration: '0.8',
        key: 'test-key',
        tags: [],
        timeEnd: 'endtimestring',
        timeStart: 'starttimestring',
      };

      expect(initialState.data).toHaveLength(0);
      const state = reducer(initialState, actions.add(payload));
      expect(state.data).toHaveLength(1);
      expect(state.data[0].key).toEqual(payload.key);
    });
  });

  describe('removing times entry', () => {
    const timeEntry = {
      key: 'test-key',
    };
    const initialState = getInitialState({
      data: [timeEntry as ITimesData],
    });

    it('should remove a times entry', () => {

      expect(initialState.data).toHaveLength(1);
      const state = reducer(initialState, actions.remove([timeEntry.key]));
      expect(state.data).toHaveLength(0);
    });
  });
});
