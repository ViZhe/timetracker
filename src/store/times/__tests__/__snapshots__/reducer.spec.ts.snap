// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Times Reducer initial state should match a snapshot 1`] = `
Object {
  "data": Array [],
  "isLoading": true,
}
`;
